policyengine_api/endpoints/policy.py

```diff
@@ -198,45 +206,66 @@ def get_policy_search(country_id: str) -> list:
     if country_not_found:
         return country_not_found
 
-    results = database.query(
-        "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
-        (country_id, f"%{query}%"),
-    )
-    if results is None:
-        return dict(
-            status="error",
-            message=f"Policy not found in {country_id}",
+    try:
+        results = database.query(
+            "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
+            (country_id, f"%{query}%"),
         )
-    else:
+        if results is None:
+            body = dict(
+                status="error",
+                message=f"Database query failed for {country_id} with query '{query}",
+            )
+            return Response(
+                json.dumps(body), status=500, mimetype="application/json"
```

Comment:
```suggestion
                json.dumps(body), status=404, mimetype="application/json"
```
See above on why I say 404 is better here than 500